[Kuah Wei Liang] iP

WeiLiangLOL opened the pull request:
As per requested by https://nus-cs2103-ay2021s2.github.io/website/admin/ip-w3.html

Commit:
Add Gradle support

Commit:
build.gradle: Update version to 8.29

WeiLiangLOL committed:
Added bot functionality

WeiLiangLOL committed:
Added listing functionality

WeiLiangLOL committed:
Added tasks

WeiLiangLOL committed:
Added different types of tasks

WeiLiangLOL committed:
Added automated testing

WeiLiangLOL committed:
Added bot exceptions

WeiLiangLOL committed:
Added delete task

WeiLiangLOL committed:
Added memory

WeiLiangLOL committed:
Added datetime

WeiLiangLOL committed:
Merge branch 'branch-Level-7'

WeiLiangLOL committed:
Merge branch 'branch-Level-8'

WeiLiangLOL committed:
Changed to follow more OOP

WeiLiangLOL committed:
Added unit tests

WeiLiangLOL committed:
Added some documentation

WeiLiangLOL committed:
Added find feature

WeiLiangLOL committed:
Merge branch 'branch-A-JavaDoc'

WeiLiangLOL committed:
Changed one line of README.md for lolz

WeiLiangLOL committed:
Merge branch 'branch-A-CodingStandard'

WeiLiangLOL committed:
Merge branch 'branch-Level-9'

WeiLiangLOL committed:
Removed .gitignore files

SoonKeatNeo commented on src/main/java/weiliang/bot/Storage.java:
```diff
@@ -0,0 +1,106 @@
+    /**
+     * The filename to store to
+     */
+    private String filename;
```
Perhaps this could be updated to fileName?

SoonKeatNeo commented on src/main/java/weiliang/bot/UI.java:
```diff
@@ -0,0 +1,193 @@
+    private String name;
+    private String logo;
+
+    private boolean active;
```
Maybe 'isActive' would be a better representation of the boolean nature of this variable?

SoonKeatNeo commented on src/main/java/weiliang/bot/UI.java:
```diff
@@ -0,0 +1,193 @@
+/**
+ * Main ui class
+ */
+public class UI {
```
Perhaps Ui or UserInterface may prevent abbreviations from being misunderstood or taken out of context.

SoonKeatNeo commented on src/main/java/weiliang/bot/task/Task.java:
```diff
@@ -0,0 +1,29 @@
+package weiliang.bot.task;
+
+public class Task {
```
I think adding header comments in the different files in your project would be helpful to aid readers to understand their main purpose and functionality :D

SoonKeatNeo commented on src/main/java/weiliang/bot/task/Task.java:
```diff
@@ -0,0 +1,29 @@
+public class Task {
+
+    protected String task;
+    protected boolean completed;
```
Perhaps this could also be updated to reflect the boolean nature?

SoonKeatNeo commented on src/main/java/weiliang/bot/task/TaskList.java:
```diff
@@ -0,0 +1,52 @@
+        if (!tasks.add(task)) {
+            throw new DukeException("Failed to add task!");
+        }
+    }
```
Perhaps it would be nice to have some JavaDocs comments reflecting the nature of each method.

SoonKeatNeo commented on src/main/java/weiliang/bot/UI.java:
```diff
@@ -0,0 +1,193 @@
+                message += "\nNo results found!";
+            }
+            return message;
+        }
```
I like the clean nature of the messages returned! Perhaps the Ui could do a bit less stuff (e.g. find) and abstract out further? (SLAP)

SoonKeatNeo reviewed (commented):
Overall, I like your code and I've learnt a few things from it about how to handle the dates and times! There are some general comments about the variable names, further abstraction and some miscellany about the JavaDocs in some of the files but otherwise, the code looks good :D.

Do let me know if I've made any mistakes or suggestions in my review!

benedictkhoomw commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,75 @@
+     */
+    public Duke() {
+        this.storage = new Storage("memory.txt");
+        this.ui = new UI("SimpleBot");
```
Consider not using `this` unless the member variable is shadowed by a parameter. I think this coding standard is not stated in the _Basic + Intermediate Rules_ page but only in the _All Rules_ page of our Java style guide.

benedictkhoomw commented on src/main/java/weiliang/bot/Storage.java:
```diff
@@ -0,0 +1,106 @@
+
+    /**
+     * Retrieves list of tasks from text document
+     * 
```
Consider ending your parameter descriptors with punctuation as specified in the coding standard.

benedictkhoomw commented on src/main/java/weiliang/bot/UI.java:
```diff
@@ -0,0 +1,193 @@
+     * @return bot formatted string response
+     * @throws DukeException if error while parsing
+     */
+    public String respond(String input) throws DukeException {
```
Not sure I like the length of this method. Maybe abstraction could be applied here. For instance, the general format seems to be 

if command matches some format, then parse the parameters, then execute the command, then return feedback

I feel that there are four responsibilities here: 
1. matching the command
2. parsing the arguments
3. executing the command
4. building the feedback message

Perhaps each responsibility could be given to a different class or method.

benedictkhoomw reviewed (commented):
Overall I like the separation of concerns present (e.g. Storage, Task) and the high level of readability of the code.

However, I feel that the UI class has too many responsibilities (for example, it seems to be handling parsing in addition to regular UI duties). Perhaps if the responsibility of parsing were given to another class it would improve the readability of your project even further.

WeiLiangLOL committed:
Merge branch 'add-gradle-support'

WeiLiangLOL committed:
Add GUI

WeiLiangLOL commented:
Hi everyone, thanks for the wonderful comments, you guys are the best! I have removed the text based UI and added a parser for command handling with the given feedback! Hoorah!

Looking forward to more reviews and thanks for tolerating my last minute 10.43PM commits!

jaredtengsw commented on src/main/java/com/weiliang/Duke.java:
```diff
@@ -0,0 +1,95 @@
+        throw new DukeException("Don't understand.");
+    }
+
+}
```
The code follows the Java Coding Standard and with good descriptive header comments. Overall, it gives the code really good readability to allow anyone to understand the implementation of the Duke class with ease.

jaredtengsw commented on build.gradle:
```diff
@@ -0,0 +1,52 @@
+    archiveClassifier = null
+}
+
+checkstyle {
```
A trivial violation of the Java Coding Standards as name is not in camel case. A quick edit should solve it!

WeiLiangLOL replied:
I wish I can resolve this, but I really don't think this is in Java (Groovy), nor could I really change anything without breaking the dependency management. Thanks though!

jaredtengsw commented on src/main/java/com/weiliang/Main.java:
```diff
@@ -0,0 +1,40 @@
+        }
+    }
+
+}
```
Code follows the Java Coding Standards and is really readable.

jaredtengsw commented on src/main/java/com/weiliang/Parser.java:
```diff
@@ -0,0 +1,102 @@
+        }
+        return message;
+    }
+}
```
Good coding style which follows the Java Coding Standards. Could consider adding header comments for the function methods to improve ease of understanding and readability of the code.

jaredtengsw commented on src/main/java/com/weiliang/task/Deadline.java:
```diff
@@ -0,0 +1,26 @@
+                + timing.format(DateTimeFormatter.ofPattern("dd MMMM YYYY, hh:mma")) + ")";
+    }
+
+}
```
Follows Java Coding Standards which is good. A suggestion would be to add header comments to methods being override to improve readability and ease of understanding especially when tenary operator is used which could make it slightly less readable.

jaredtengsw commented on src/main/java/com/weiliang/task/Event.java:
```diff
@@ -0,0 +1,26 @@
+                + timing.format(DateTimeFormatter.ofPattern("dd MMMM YYYY, hh:mma")) + ")";
+    }
+
+}
```
Similar to Deadline Class. The code follows Java Coding Standards and  a suggestion would be to add header comments to methods being override.

jaredtengsw reviewed (commented):
Your code is really good and readable. A suggestion from me would be to consider adding header comments to some methods to improve ease of understanding but it's just a suggestion and it's fine even without them.

WeiLiangLOL committed:
Add assertions

WeiLiangLOL committed:
Add more javadoc comments

WeiLiangLOL committed:
Merge pull request #1 from WeiLiangLOL/branch-A-Assertions

Add assertions

WeiLiangLOL committed:
Merge branch 'master' into branch-A-CodeQuality

WeiLiangLOL committed:
Merge pull request #2 from WeiLiangLOL/branch-A-CodeQuality

Add code quality

WeiLiangLOL committed:
Add undo

WeiLiangLOL committed:
Merge branch 'master' of https://github.com/WeiLiangLOL/ip

WeiLiangLOL committed:
Add user guide

WeiLiangLOL committed:
Add more javadoc

WeiLiangLOL committed:
Added final test case

WeiLiangLOL committed:
Fix import ordering

WeiLiangLOL committed:
Update commands

WeiLiangLOL committed:
Add compatibility to 1.8

WeiLiangLOL committed:
Fix runtime for linux and mac
